### PR TITLE
add gas conditional if simple transaction

### PR DESCRIPTION
**Pull request by luu-alex**
## Description
#7021 
There seems to be a bug when using sendTransaction with no gas property in the rpc method eth_call in `checkRevertBeforeSending`. Testing with both metamask and an infura provider, with a simple transaction (with no data or input so no contract call)
eth_call will revert and say there is insufficient funds when there should be easily enough funds to cover the gas and value. 
The transaction looks like 
```
const tx = {
        to: 0,
        value:1,
        from: accounts[0].address,
    }
```
```
InvalidResponseError: Returned error: err: insufficient funds for gas * price + value: address 0xBe1320eA9Fd1b34C0cf664eFbb29CF2dBD3cC81b have 100000000000000000 want 2859580732500000001 (supplied gas 50000000)
    at Web3RequestManager._processJsonRpcResponse (/Users/alexluu/practice/v4web3js/node_modules/web3/node_modules/web3-core/lib/commonjs/web3_request_manager.js:258:23)
    at Web3RequestManager.<anonymous> (/Users/alexluu/practice/v4web3js/node_modules/web3/node_modules/web3-core/lib/commonjs/web3_request_manager.js:156:29)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/alexluu/practice/v4web3js/node_modules/web3/node_modules/web3-core/lib/commonjs/web3_request_manager.js:21:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  innerError: {
    code: -32000,
    message: 'err: insufficient funds for gas * price + value: address 0xBe1320eA9Fd1b34C0cf664eFbb29CF2dBD3cC81b have 100000000000000000 want 2859580732500000001 (supplied gas 50000000)'
```  

## Solution

Within `checkRevertBeforeSending`, A simple fix to this is to check if the transaction does not have data set we explicitly set the gas to 21000  

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](https://github.com/web3/web3.js/blob/4.x/.github/CONTRIBUTING.md).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3-eth/src/utils/send_tx_helper.ts [hunk 1]
```diff
@@ -122,7 +122,14 @@ export class SendTxHelper<
 
 	public async checkRevertBeforeSending(tx: TransactionCall) {
 		if (this.options.checkRevertBeforeSending !== false) {
-			const reason = await getRevertReason(this.web3Context, tx, this.options.contractAbi);
+			let formatTx = tx;
+			if (isNullish(tx.data) && isNullish(tx.input) && isNullish(tx.gas)) { // eth.call runs into error if data isnt filled and gas is not defined, its a simple transaction so we fill it with 21000
```
Comment (jdevcs) [inline packages/web3-eth/src/utils/send_tx_helper.ts:126]: when data is populated, is gas limit always populated in that case?

Comment (jdevcs) [reply packages/web3-eth/src/utils/send_tx_helper.ts:126]: and its only missing in case of value transfers `eth_calls` only?

**Review by jdevcs — APPROVED** [review]

File: packages/web3-eth/src/utils/send_tx_helper.ts (hunk 1 above)
Comment (jdevcs) [reply packages/web3-eth/src/utils/send_tx_helper.ts:126]: additionally, if tx has gas limit only for eth_call rpc( for revert check) and not in actual Tx that will be sent after it, 
eth_call may pass while actual Tx may fail in some cases, as both will differ, so I think gas limit should be applied in all cases

Comment (luu-alex) [reply packages/web3-eth/src/utils/send_tx_helper.ts:126]: eth_call with no data and no gas will fail from provider, but send_transaction gas will be filled automatically. i belive this is just a weird edge case, and with our tests passing i dont believe this will be an issue.
